cli/dcoscli/package/main.py

```diff
@@ -206,6 +215,235 @@ def _remove_repo(repo_name):
+    """ Resolves all local references in a DC/OS Package Build Definition
+
+    :param build_definition: The DC/OS Package Build Definition that may
+     contain local references
+    :type build_definition: dict
+    :param build_definition_directory: The directory of the Build Definition
+    :type build_definition_directory: str
+    :param build_schema: The schema for the Build Definition
+    :type build_schema: dict
+    """
+    _replace_marathon(build_definition,
+                      build_schema,
+                      build_definition_directory)
+
+    _replace_directly(build_definition,
+                      build_schema,
+                      build_definition_directory,
+                      "config")
+
+    _replace_directly(build_definition,
+                      build_schema,
+                      build_definition_directory,
+                      "resource")
+
+
+def _replace_directly(build_definition,
+                      build_schema,
+                      build_definition_directory,
+                      ref):
+    """ Replaces the local reference ref with the contents of
+     the file pointed to by ref
+
+    :param build_definition: The DC/OS Package Build Definition that
+    may contain local references
+    :type build_definition: dict
+    :param build_definition_directory: The directory of the Build Definition
+    :type build_definition_directory: str
+    :param build_schema: The schema for the Build Definition
+    :type build_schema: dict
+    :param ref: The key in build_definition that will be replaced
+    :type ref: str
+    """
+    if ref in build_definition and _is_local_reference(build_definition[ref]):
+        location = build_definition[ref][1:]
+        if not os.path.isabs(location):
+            location = os.path.join(build_definition_directory, location)
+
+        with util.open_file(location) as f:
+            contents = util.load_json(f, True)
+
+        build_definition[ref] = contents
+
+        errs = util.validate_json(build_definition, build_schema)
+        if errs:
+            raise DCOSException('Error validating package: '
+                                '[{}] does not conform to the'
+                                ' specified schema'.format(location))
+
+
+def _replace_marathon(build_definition,
+                      build_schema,
+                      build_definition_directory):
+    """ Replaces the marathon v2AppMustacheTemplate ref with
+     the base64 encoding of the file pointed to by the reference
+
+    :param build_definition: The DC/OS Package Build Definition that
+     may contain local references
+    :type build_definition: dict
+    :param build_definition_directory: The directory of the Build Definition
+    :type build_definition_directory: str
+    :param build_schema: The schema for the Build Definition
+    :type build_schema: dict
+    """
+    ref = "marathon"
+    tp = "v2AppMustacheTemplate"
```

Comment:
@tamarrow tp stands for template, as in v2AppMustancheTemplate. It's just there so that I don't have to type long names.

What is it is { 'marathon': { 'v2AppMustacheTemplate': 'euhantoeuhsnaoh' }}